### PR TITLE
fix(core): fix panic on nil pointer dereference by passing KAS the SDK instance on registration

### PR DESCRIPTION
**Pull request by jakedoublev**

### CODE REVIEW

File: service/kas/kas.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func NewRegistration() serviceregistry.Registration {
 		ServiceDesc: &kaspb.AccessService_ServiceDesc,
 		RegisterFunc: func(srp serviceregistry.RegistrationParams) (any, serviceregistry.HandlerServer) {
 			// FIXME msg="mismatched key access url" keyAccessURL=http://localhost:9000 kasURL=https://:9000
```
Comment (jakedoublev) [inline service/kas/kas.go:60]: @pflynn-virtru was there an issue to track this?

File: service/kas/kas.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func NewRegistration() serviceregistry.Registration {
...
-			kasURLString := "https://" + srp.OTDF.HTTPServer.Addr
+			kasURLString := "http://localhost" + srp.OTDF.HTTPServer.Addr
```
Comment (dmihalcik-virtru) [inline service/kas/kas.go:61]:
```suggestion
			hostWithPort := srp.OTDF.HTTPServer.Addr
			if strings.HasPrefix(hostWithPort, ":") {
				hostWithPort = "localhost" + hostWithPort
			}
			kasURLString := "http://" + hostWithPort
```

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]

File: service/kas/kas.go (hunk 2 above)
Comment (dmihalcik-virtru) [reply service/kas/kas.go:61]: We probably should also somehow make the scheme (http or https) configurable

**Review by dmihalcik-virtru — APPROVED** [review]

File: service/kas/kas.go (hunk 1 above)
Comment (dmihalcik-virtru) [reply service/kas/kas.go:61]: I've filed #578. Feel free to comment there with more details, @jakedoublev or @pflynn-virtru

**Pull request merged**